src/gtk/textctrl.cpp

```diff
@@ -728,7 +728,26 @@ bool wxTextCtrl::Create( wxWindow *parent,
         gtk_entry_get_text((GtkEntry*)m_text);
 
         if (style & wxNO_BORDER)
-            g_object_set (m_text, "has-frame", FALSE, NULL);
+        {
+#ifdef __WXGTK3__
+            // this is sort of a workaround for when the builtin theme
+            // won't remove the frame by itself -- see
+            // https://bugzilla.gnome.org/show_bug.cgi?id=789732#c1
+            GtkCssProvider* provider = gtk_css_provider_new();
+            gtk_css_provider_load_from_data(
+                provider,
+                gtk_check_version(3,20,0) ? "entry" : "GtkEntry"
+                " { border: none; border-radius: 0; padding: 0 }",
+                -1, NULL);
+            gtk_style_context_add_provider(
+                gtk_widget_get_style_context(m_widget),
+                GTK_STYLE_PROVIDER(provider),
+                GTK_STYLE_PROVIDER_PRIORITY_USER);
```

Comment:
I don't think this is the right priority to use, `GTK_STYLE_PROVIDER_PRIORITY_APPLICATION` is probably right.